### PR TITLE
Key by ['locus', 'alleles'] in import_vcfs

### PR DESCRIPTION
**Pull request by chrisvittal**
Also fixup the combiner to work properly with the new keys.

### CODE REVIEW

**Comment by chrisvittal** [comment]
The changes to the combiner have mostly no basis on the changes to import_vcfs. I can remove them and PR separately if you would like.

File: hail/python/hail/experimental/vcf_combiner.py [hunk 1]
```diff
@@ -97,15 +88,19 @@ def combine_gvcfs(mts):
     def localize(mt):
         return mt._localize_entries('__entries', '__cols')
 
-    cols = None
-    for mt in mts:
-        if cols is None:
```
Comment (jigold) [inline hail/python/hail/experimental/vcf_combiner.py]: What was this code supposed to be doing before? It looks like it was completely deleted.

File: hail/python/hail/experimental/vcf_combiner.py [hunk 2]
```diff
@@ -97,15 +88,19 @@ def combine_gvcfs(mts):
...
-        else:
-            cols = cols.union(mt.key_cols_by().cols())
+    mts = [hl.MatrixTable(MatrixKeyRowsBy(mt._mir, ['locus'], is_sorted=True)) for mt in mts]
+    mts = [mt.annotate_rows(
+        alleles=hl.bind(lambda ref: mt.alleles[1:].map(lambda alt: hl.struct(ref=ref, alt=alt)),
+                        mt.alleles[0])) for mt in mts]
     ts = hl.Table._multi_way_zip_join([localize(mt) for mt in mts], 'data', 'g')
     combined = combine(ts)
-    return combined._unlocalize_entries('__entries', '__cols', ['s'])
+    combined = combined.annotate(alleles=combined.alleles[:1].map(lambda d: d.ref)
```
Comment (jigold) [inline hail/python/hail/experimental/vcf_combiner.py]: Nice!

File: hail/python/hail/experimental/vcf_combiner.py (hunk 1 above)
Comment (chrisvittal) [reply hail/python/hail/experimental/vcf_combiner.py]: This code was needed before CastMatrixTableToTable kept the column information. We never use `cols`, so the code became dead.

**Pull request merged**